do/serverless.go

```diff
@@ -745,6 +783,100 @@ func (s *serverlessService) WriteProject(project ServerlessProject) (string, err
+// string it is assumed to be the package-qualified name of a function and only the triggers
+// of that function are listed.  If 'fcn' is empty all triggers are listed.
+func (s *serverlessService) ListTriggers(ctx context.Context, fcn string) ([]ServerlessTrigger, error) {
+	empty := []ServerlessTrigger{}
+	creds, err := s.ReadCredentials()
+	if err != nil {
+		return empty, err
+	}
+	path := "v2/functions/triggers/" + creds.Namespace
+	req, err := s.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return empty, err
+	}
+	decoded := new(ServerlessTriggerListResponse)
+	_, err = s.client.Do(ctx, req, decoded)
+	if err != nil {
+		return empty, err
+	}
+	triggers := decoded.Triggers
+	// The API does not filter by function; that is done here.
+	if fcn != "" {
+		filtered := []ServerlessTrigger{}
+		for _, trigger := range triggers {
+			if trigger.Function == fcn {
+				filtered = append(filtered, trigger)
+			}
+		}
+		triggers = filtered
+	}
+	return fixBaseDates(triggers), nil
+}
+
+// fixBaseDates applies fixBaseDate to an array of triggers
+func fixBaseDates(list []ServerlessTrigger) []ServerlessTrigger {
+	ans := []ServerlessTrigger{}
+	for _, trigger := range list {
+		ans = append(ans, fixBaseDate(trigger))
+	}
+	return ans
+}
+
+// fixBaseDate fixes up the LastRun field of a trigger that has never been run.
+// It should properly contain blank but sometimes contain an encoding of the base date (a string
+// starting with "000").
+func fixBaseDate(trigger ServerlessTrigger) ServerlessTrigger {
+	if strings.HasPrefix(trigger.LastRun, "000") {
+		trigger.LastRun = "_"
+	}
+	return trigger
+}
+
+// GetTrigger gets the contents of a trigger for display
+func (s *serverlessService) GetTrigger(ctx context.Context, name string) (ServerlessTrigger, error) {
+	empty := ServerlessTrigger{}
+	creds, err := s.ReadCredentials()
+	if err != nil {
+		return empty, err
+	}
+	path := "v2/functions/trigger/" + creds.Namespace + "/" + name
+	req, err := s.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return empty, err
+	}
+	decoded := new(ServerlessTriggerGetResponse)
+	_, err = s.client.Do(ctx, req, decoded)
+	if err != nil {
+		return empty, err
+	}
+	return fixBaseDate(decoded.Trigger), nil
+}
+
+// Delete Trigger deletes a trigger from the namespace (used when undeploying triggers explicitly,
+// not part of a more general undeploy; when undeploying a function or the entire namespace we rely
+// on the deployer to delete associated triggers).
+func (s *serverlessService) DeleteTrigger(ctx context.Context, name string) error {
+	creds, err := s.ReadCredentials()
+	if err != nil {
+		return err
+	}
+	path := "v2/functions/trigger/" + creds.Namespace + "/" + name
+	_, err = s.client.NewRequest(ctx, http.MethodDelete, path, nil)
+	return err
+}
+
+// fixupCron detects the optional seconds field and removes it
+func fixupCron(cron string) string {
+	parts := strings.Split(cron, " ")
```

Comment:
Seconds are not supported externally but the scheduler adds a random seconds field when storing the task in order to avoid bursts.   In the development of the various API layers sometimes we. were getting back 6-element cron expressions (in which the first field, seconds, was misleading) and sometimes these were already trimmed.   The code here will work either way and either way the seconds are not displayed.